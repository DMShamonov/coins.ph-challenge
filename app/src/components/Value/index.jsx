// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PropsType } from './types.js.flow';

export default function Value({
  value,
}: PropsType): React.Node {
  return (
    <p
      className={classnames('value', {
        value_is_negative: value < 0,
        value_is_positive: value > 0,
      })}
    >
      {value}
    </p>
  );
}
