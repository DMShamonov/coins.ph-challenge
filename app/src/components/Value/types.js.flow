// @flow

export type PropsType = {|
  value: number,
|};
