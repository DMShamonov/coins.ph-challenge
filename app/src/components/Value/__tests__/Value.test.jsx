import React from 'react';
import { shallow } from 'enzyme';

import Value from '../index';

describe('`Value` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Value value={0} />)).toMatchSnapshot();
  });

  test('when `value` prop is positive', () => {
    expect(shallow(<Value value={10} />)).toMatchSnapshot();
  });

  test('when `value` prop is negative', () => {
    expect(shallow(<Value value={-10} />)).toMatchSnapshot();
  });
});
