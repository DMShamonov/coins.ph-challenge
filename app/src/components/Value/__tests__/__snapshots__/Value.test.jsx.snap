// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Value\` component should render correctly when \`value\` prop is negative 1`] = `
<p
  className="value value_is_negative"
>
  -10
</p>
`;

exports[`\`Value\` component should render correctly when \`value\` prop is positive 1`] = `
<p
  className="value value_is_positive"
>
  10
</p>
`;

exports[`\`Value\` component should render correctly with default props 1`] = `
<p
  className="value"
>
  0
</p>
`;
